### DISCUSSION TITLE
How to ignore concurrency for manual runs?

### DISCUSSION THREAD

**Discussion by vitalyk-multinarity** [community/community#31908] [Actions] [answered]
I'm using 

```
concurrency:
  group: ${{ github.workflow }}-${{ github.head_ref || github.ref_name }}
  cancel-in-progress: true
```

my triggers are pull, push events, and workflow_dispatch. 
I want to keep concurrency for automatic triggers, but ignore for manual trigger. How can I reach that?

**Comment by MathiasReker** [comment, marked as answer]
Check this example: https://github.com/MathiasReker/Java-CI-CD/blob/develop/.github/workflows/ci-cd.yml

line 68:     `if: ${{ github.event_name != 'pull_request' }}`

**Reply by vitalyk-multinarity** [reply]
Thank you, but I'm not sure I understand how I can use this or a similar expression to deactivate concurrency for manual builds.

**Comment by Jeffwhen** [comment]
Try

```
concurrency:
  group: ${{ github.head_ref || github.run_id }}
  cancel-in-progress: true
```

From [github doc](https://docs.github.com/en/actions/using-jobs/using-concurrency#example-using-a-fallback-value).

**Comment by vitalyk-multinarity** [comment]
Sorry, I didn't update my issue: at the end I'm using the following expression:

`group: ${{ github.workflow }}-${{ github.head_ref || github.ref_name }}-${{ github.event_name == 'workflow_dispatch' && github.run_id }}`
